### PR TITLE
Role additions and updates to standard install profile

### PR DESCRIPTION
**Pull request by ben-hamelin**
## Summary

Adds 3 new custom roles, and makes some minor updates to existing and "authenticated" role.
Adds some local development documentation regarding feature branches as well.

## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | yes
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | no (already covered)
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | JIRA issue, bug reports, security alerts, etc.
https://thinkoomph.jira.com/browse/RIG-15

### CODE REVIEW

File: ecms_base/config/install/user.role.drupal_administrator.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+langcode: en
+status: true
+dependencies: {  }
+id: drupal_administrator
+label: 'Drupal Administrator'
+weight: 3
+is_admin: null
```
Comment (pfrilling) [inline ecms_base/config/install/user.role.drupal_administrator.yml:7]: We should probably mark this role as an admin, so it inherits all permissions by default. Maybe we should rename the existing `administrator` role to this instead?

File: ecms_base/config/install/user.role.authenticated.yml [hunk 2]
```diff
@@ -6,6 +6,15 @@ label: 'Authenticated user'
 weight: 1
 is_admin: false
 permissions:
+  - 'access administration pages'
```
Comment (pfrilling) [inline ecms_base/config/install/user.role.authenticated.yml:9]: Are there any security implications here by giving the authenticated user these administrative permissions? Should we move these to the elevated author/editor roles instead? I _think_ it is fine, but wanted to flag it anyway.

File: ecms_base/config/install/user.role.content_editor.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+langcode: en
+status: true
+dependencies: {  }
+id: content_editor
+label: 'Content Editor'
```
Comment (pfrilling) [inline ecms_base/config/install/user.role.content_editor.yml:5]: Should this be named `Content Author` instead to match the roles & workflows documentation?

**Review by pfrilling — CHANGES_REQUESTED** [review]
Looks good. I just had a couple changes/questions. Also, do we want to include all of the roles in this PR?

File: ecms_base/config/install/user.role.drupal_administrator.yml (hunk 1 above)
Comment (ben-hamelin) [reply ecms_base/config/install/user.role.drupal_administrator.yml:7]: Got it, going to update to match the roles & publishing workflows document.

File: ecms_base/config/install/user.role.authenticated.yml (hunk 2 above)
Comment (ben-hamelin) [reply ecms_base/config/install/user.role.authenticated.yml:9]: I will test how removing this impacts the content author role and adjust accordingly

File: ecms_base/config/install/user.role.administrator.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ langcode: en
 status: true
 dependencies: {  }
 id: administrator
```
Comment (pfrilling) [inline ecms_base/config/install/user.role.administrator.yml:4]: I think we should rename the id to `drupal_admin` to be consistent. The filename should also reflect that change.

We will want to update the `ecms_base.install` file to update the role assignment for user/1 too.

**Review by pfrilling — CHANGES_REQUESTED** [review]
One quick change to the admin role configuration.

**Review by pfrilling — APPROVED** [review]
I think this looks good. 👍

**Pull request merged**